e2e/test/Helpers/logging/TestLogger.cs

```diff
@@ -89,14 +76,14 @@ public void Event(string eventName, IDictionary<string, string> extraProperties
         /// <summary>
         /// Flush to ensure logs are not lost before quiting an application.
         /// </summary>
-        public void SafeFlush()
+        public async Task SafeFlushAsync()
         {
             if (_telemetryClient != null)
             {
                 _telemetryClient.Flush();
 
                 // Hold on to the thread context to allow flushing to complete.
-                Thread.Sleep(5000);
+                await Task.Delay(5000).ConfigureAwait(false);
```

Comment:
I added a task delay instead of a thread sleep. @vinagesh - is this ok, or does this need to be a thread sleep?

Reply:
No we should not use Thread.Sleep at all. I missed it. It should always be Task.Delay.